src/js/service/CodeOrgMessageService.js

```diff
@@ -0,0 +1,72 @@
+/* Code.org-specific service for managing communication with the outer page
+ * when used as an embedded app in an iframe. */
+(function () {
+  var ns = $.namespace('pskl.service');
+  var MessageType = ExportedConstants.MessageType;
+
+  /**
+   * @param {!ImportService} importService
+   * @constructor
+   */
+  ns.CodeOrgMessageService = function (importService) {
+    /**
+     * The import service we'll rely on to load animations when they are
+     * selected in the outer UI.
+     * @private {!ImportService}
+     */
+    this.importService_ = importService;
+
+    /**
+     * Which origin we will receive messages from.  Set on init().
+     * @private {string}
+     */
+    this.allowedOrigin_ = null;
+  };
+
+  /**
+   * Initialize message service.  Causes service to start listening for messages
+   * on window.
+   * @param {!Window} window
+   */
+  ns.CodeOrgMessageService.prototype.init = function (window) {
```

Comment:
I wonder if it would be better to name the param something like `originWindow` if it's able to be a different window object than the global window.